### PR TITLE
Egg Changes

### PR DESCRIPTION
**Pull request by Zonespace27**
# About the pull request
Now that https://github.com/cmss13-devs/cmss13/pull/4716 's been merged, here's the changes I've wanted to make for eggs for a bit now.

- Eggs can no longer be planted in vehicles
- Eggs can no longer be planted on tiles that would obstruct the egg. 

# Explain why it's good for the game

- In the vast majority of cases, you cannot tell that a vehicle you're entering is infested with you-instantly-lose eggs. While normally getting hit with an egg is reasonably predictable (there are weeds == possibility of eggs), having no indication of it until you go inside and get hugged isn't fun or balanced.
- Eggs being able to layer under things like grass is technically a "feature", but not one I would like to keep. Traps serve the exact same purpose, but are much harder to mass produce (as intended). Eggs should be cheaper but more detectable, which doesn't align them being able to be near-fully hidden.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

image

</details>

# Changelog
:cl:
balance: You cannot plant eggs inside vehicles.
balance: Eggs can no longer be planted on tiles with objects that would obscure them.
/:cl:

### CODE REVIEW

**Review by fira — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/egg_item.dm [hunk 1]
```diff
@@ -106,6 +106,12 @@
 		to_chat(user, SPAN_XENOWARNING("[src] cannot be planted inside a vehicle."))
 		return
 
+	for(var/obj/object in T.contents)
+		var/obj/effect/alien/egg/xeno_egg = /obj/effect/alien/egg
+		if(object.layer > initial(xeno_egg.layer))
+			to_chat(user, SPAN_XENOWARNING("[src] cannot be planted below objects that would obscure it."))
```
Comment (fira) [inline code/modules/mob/living/carbon/xenomorph/egg_item.dm:112]: Ehh i don't know, what if the object gets put after? I wasn't a fan of previous one but i think it made more sense

**Comment by Releasethesea** [comment]
since this is TM'd i assume i can leave feedback on it, so far my experience has been very inconsistent i can place eggs in some corners and some others i cant ive also not been able to place eggs when there is a railing next to them, aswell as from what ive seen papers and other items on the floors, like roller chairs

File: code/modules/mob/living/carbon/xenomorph/egg_item.dm (hunk 1 above)
Comment (Zonespace27) [reply code/modules/mob/living/carbon/xenomorph/egg_item.dm:112]: Missed this earlier; I don't think it'll be a problem, there aren't many things that'd layer over an egg's sprite that xenoes can create or move.

**Review by Birdtalon — APPROVED** [review]
TM has been going quite a while with no obvious unintended effects.

**Pull request merged**